variables.mk

```diff
@@ -17,19 +18,39 @@
 # 	SUB_PROJECT = use the specific subproject default variables
 #########################################################################################
 PROJECT     ?= example
-MODEL       ?= TestHarness
-CONFIG      ?= DefaultExampleConfig
+MODEL       ?= RocketTestHarness
+VLOG_MODEL  ?= TestHarness
+CONFIG      ?= DefaultRocketConfig
 CFG_PROJECT ?= $(PROJECT)
 SBT_PROJECT ?= $(PROJECT)
 TB          ?= TestDriver
-TOP         ?= ExampleTop
+TOP         ?= RocketTop
 
+# make it so that you only change 1 param to change most or all of them!
 SUB_PROJECT ?= example
-ifeq ($(SUB_PROJECT),boom) # make it so that you only change 1 param to change them all!
-	SBT_PROJECT=boom
+ifeq ($(SUB_PROJECT),boomexample)
+	# for a BOOM based system (provides all necessary params)
+	MODEL=BoomTestHarness
+	CONFIG=DefaultBoomConfig
+	TOP=BoomTop
+endif
+ifeq ($(SUB_PROJECT),boom)
+	# for BOOM developers (only need to provide a CONFIG)
 	PROJECT=boom.system
+	MODEL=TestHarness
+	CFG_PROJECT=boom.system
+	SBT_PROJECT=boom
 	TOP=ExampleBoomSystem
 endif
+ifeq ($(SUB_PROJECT),rocketchip)
+	# for Rocket-chip developers
+	PROJECT=freechips.rocketchip.system
+	MODEL=TestHarness
+	CONFIG=DefaultConfig
```

Comment:
I know I'm too late but is it not worth letting the user set `CONFIG` for all subprojects but with sane defaults using `?=`?

Reply:
True. I think the best thing to do is set the other defaults and leave `CONFIG` unset

Reply:
Did #71 address your concerns?

Reply:
Yes.